packages/create-astro/test/create-astro.test.js

```diff
@@ -1,46 +1,120 @@
 import fs from 'fs';
 import path from 'path';
 import { fileURLToPath } from 'url';
+import http from 'http';
 import { suite } from 'uvu';
 import execa from 'execa';
 import del from 'del';
+import glob from 'tiny-glob';
 import * as assert from 'uvu/assert';
 import { TEMPLATES } from '../dist/templates.js';
 
-const CreateAstro = suite('npm init astro');
+// config
+const GITHUB_SHA = process.env.GITHUB_SHA || execa.sync('git', ['rev-parse', 'HEAD']).stdout; // process.env.GITHUB_SHA will be set in CI; if testing locally execa() will gather this
+const MAX_TEST_TIME = 60000; // maximum time a test may take (60s)
+const TIMER = {}; // keep track of every test’s run time (uvu requires manual setup for this)
+const FIXTURES_DIR = path.join(fileURLToPath(path.dirname(import.meta.url)), 'fixtures');
 
-const cwd = fileURLToPath(path.dirname(import.meta.url));
-const fixturesDir = path.join(cwd, 'fixtures');
+// helper
+async function fetch(url) {
+  return new Promise((resolve) => {
+    http
+      .get(url, (res) => {
+        let body = '';
+        res.on('data', (chunk) => {
+          body += chunk;
+        });
+        res.on('end', () => resolve({ statusCode: res.statusCode, body }));
+      })
+      .on('error', (err) => {
+        reject(err);
+      });
+  });
+}
+
+// test
+const CreateAstro = suite('npm init astro');
 
 CreateAstro.before(async () => {
-  await del(fixturesDir);
-  await fs.promises.mkdir(fixturesDir);
+  // clean install dir
+  await del(FIXTURES_DIR);
+  await fs.promises.mkdir(FIXTURES_DIR);
+
+  // install all templates & deps before running tests
+  await Promise.all(
+    TEMPLATES.map(async ({ value: template }) => {
+      const templateDir = path.join(FIXTURES_DIR, template);
+      await execa('../../create-astro.mjs', [templateDir, '--template', template, '--commit', GITHUB_SHA, '--force-overwrite'], {
+        cwd: FIXTURES_DIR,
+      });
+      await execa('yarn', ['--frozen-lockfile', '--silent'], { cwd: templateDir });
+    })
+  );
 });
 
-for (const { value: template } of TEMPLATES) {
-  // TODO: Unskip once repo is made public. Because the repo is private, the templates can't yet be downloaded.
-  CreateAstro.skip(template, async () => {
-    const testDirectory = path.join(fixturesDir, template);
-    const { stdout } = await execa('../../create-astro.mjs', [testDirectory, '--template', template, '--force-overwrite'], { cwd: path.join(cwd, 'fixtures') });
+// enforce MAX_TEST_TIME
+CreateAstro.before.each(({ __test__ }) => {
+  if (TIMER[__test__]) throw new Error(`Test "${__test__}" already declared`);
+  TIMER[__test__] = setTimeout(() => {
+    throw new Error(`"${__test__}" did not finish within allowed time`);
```

Comment:
Because `uvu` won’t ever finish if there are open threads, this test is super probable to run indefinitely. We had to add the same manual timeout handling here as we do in the astro tests.